mlir/lib/Dialect/MemRef/IR/MemRefOps.cpp

```diff
@@ -2237,6 +2237,17 @@ FailureOr<MemRefType> ExpandShapeOp::computeExpandedType(
                          srcType.getMemorySpace());
 }
 
+LogicalResult ExpandShapeOp::inferOutputShape(
+    OpBuilder &b, Location loc, MemRefType expandedType,
+    ArrayRef<ReassociationIndices> reassociation,
+    ArrayRef<OpFoldResult> inputShape,
+    std::pair<SmallVector<int64_t>, SmallVector<Value>> &outputShape) {
+  auto expandedTensorType =
```

Comment:
Note: This triggered the suggestion of using `ShapedType` in `inferExpandShapeOutputShape` to avoid having to do this.